nemo_skills/inference/model/code_execution.py

```diff
@@ -321,153 +278,27 @@ def generate_async(
-
-        Not every server supports that, so make sure to override this method directly if that's not the case.
-        """
-        generation_ids = self.generate_async(
-            prompts=prompts,
-            code_begin=code_begin,
-            code_end=code_end,
-            code_output_begin=code_output_begin,
-            code_output_end=code_output_end,
-            code_output_format=code_output_format,
-            tokens_to_generate=tokens_to_generate,
-            temperature=temperature,
-            top_p=top_p,
-            top_k=top_k,
-            min_p=min_p,
-            repetition_penalty=repetition_penalty,
-            random_seed=random_seed,
-            stop_phrases=stop_phrases,
-            remove_stop_phrases=remove_stop_phrases,
-            top_logprobs=top_logprobs,
-            timeout=timeout,
-            max_code_executions=max_code_executions,
-            stream=stream,
-            extra_body=extra_body,
-        )
-        all_generations = [None] * len(prompts)
-        while True:
-            remaining_ids = [generation_id for generation_id in generation_ids if generation_id is not None]
-            if len(remaining_ids) == 0:
-                break
-            remaining_positions = [
-                idx for idx, generation_id in enumerate(generation_ids) if generation_id is not None
-            ]
-            generations = self.get_generations(remaining_ids)
-            for gen_pos, gen_dict in zip(remaining_positions, generations):
-                if isinstance(gen_dict, Generator) or gen_dict['generation'] is not None:  # will be None until done
-                    generation_ids[gen_pos] = None
-                    all_generations[gen_pos] = gen_dict
-
-            time.sleep(1)
-
-        return all_generations
-
-    async def generate_asyncio(self, *args, **kwargs) -> dict:
-        result = await asyncio.to_thread(self.generate, *args, **kwargs)
-        assert len(result) == 1, "generate_asyncio should return a single result"
-        return result[0]
+        
+        request = {key: value for key, value in kwargs.items()}
+        request['prompt'] = prompt
+        self.model.preprocess_request(request)
+        
+        output = self._generate_single(**request)
+        
+        if remove_stop_phrases:
+            stop_phrases_for_request = request['stop_phrases']
+            if output['generation'] is not None:
+                output['generation'] = trim_after_stop_phrases(output['generation'], stop_phrases_for_request)
+        
+        return output
+
+    async def generate_asyncio(self, prompt, *args, **kwargs) -> dict:
+        # Configure the executor for the current event loop
+        loop = asyncio.get_running_loop()
+        if not hasattr(loop, '_nemo_skills_executor_configured'):
+            loop.set_default_executor(ThreadPoolExecutor(max_workers=2048))
```

Comment:
Maybe add `max_workers` to `CodeExecutionConfig`.

And do

```suggestion
            loop.set_default_executor(ThreadPoolExecutor(max_workers=self.config.max_workers))
```

Reply:
Oh, I just want it to be configurable. `max_concurrent_requests` should work as well.